update_version.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -e
+
+# Use this script to automate updating pki version.
+#
+# Usage: ./update_version.sh <major> <minor> <update> <phase> # (phase is optional)
+#
+# Explanation:
+# -    change_spec_version
+# -        Updates the spec version to the new version provided
+# -    commit_version_change
+# -        Commits that change
+# -    create_tag
+# -        Creates a tag based on the new version provided
+# -    create_source_tarball
+# -        Creates a source tarball based on the new version provided
+
+NEXT_MAJOR=$1
+NEXT_MINOR=$2
+NEXT_UPDATE=$3
+NEXT_PHASE=$4
+
+if [ -z "$NEXT_PHASE" ] ; then
+    NEXT_VERSION=$NEXT_MAJOR.$NEXT_MINOR.$NEXT_UPDATE
+else
+    NEXT_VERSION=$NEXT_MAJOR.$NEXT_MINOR.$NEXT_UPDATE-$NEXT_PHASE
+fi
+echo "New version is $NEXT_VERSION"
+
+verify_phase() {
+    if [[ "$NEXT_PHASE" =~ ^(alpha|beta)[0-9]+$ ]] ; then
+        echo "$NEXT_PHASE is a valid phase"
+    elif [ -z "$NEXT_PHASE" ] ; then
+        echo "Empty phase"
+    else
+        echo "$NEXT_PHASE is an invalid phase, aborting"
+        exit 1
+    fi
+}
+
+change_spec_version() {
+    CURRENT_MAJOR=$(grep "major_version " pki.spec | grep -Eo '[0-9]+$')
+    CURRENT_MINOR=$(grep "minor_version " pki.spec | grep -Eo '[0-9]+$')
+    CURRENT_UPDATE=$(grep "update_version " pki.spec | grep -Eo '[0-9]+$')
+    CURRENT_PHASE=$(grep "phase " pki.spec | grep -E 'alpha|beta' | awk '{print $(NF)}')
+
+    if [[ "$CURRENT_MAJOR" != "$NEXT_MAJOR" ]] ; then
+       echo "Update major version to $NEXT_MAJOR"
+       sed -i "/major_version /c\%global           major_version $NEXT_MAJOR" pki.spec
+    fi
+
+    if [[ "$CURRENT_MINOR" != "$NEXT_MINOR" ]] ; then
+        echo "Update minor version to $NEXT_MINOR"
+        sed -i "/minor_version /c\%global           minor_version $NEXT_MINOR" pki.spec
+    fi
+
+    if [[ "$CURRENT_UPDATE" != "$NEXT_UPDATE" ]] ; then
+        echo "Update update version to $NEXT_UPDATE"
+        sed -i "/update_version /c\%global           update_version $NEXT_UPDATE" pki.spec
+    fi
+
+    if [[ "$CURRENT_PHASE" != "$NEXT_PHASE" ]] ; then
+        if [ -z "$NEXT_PHASE" ] ; then
+            echo "Remove phase"
+            sed -i "/phase /c\#global           phase" pki.spec
+            echo "Update release_number"
+            sed -i "/release_number /c\%global           release_number 1" pki.spec
+        elif [ -z "$CURRENT_PHASE" ] ; then
+            echo "Add phase, set to $NEXT_PHASE"
+            sed -i "/#global           phase/c\%global           phase $NEXT_PHASE" pki.spec
+            echo "Update release_number"
+            sed -i "/release_number /c\%global           release_number 0.1" pki.spec
+        else
+            echo "Update phase to $NEXT_PHASE"
+            sed -i "/phase /c\%global           phase $NEXT_PHASE" pki.spec
+            echo "Update release_number"
+            sed -i "/release_number /c\%global           release_number 0.1" pki.spec
```

Comment:
Ideally if we update the phase we should bump the release number too so that the whole NVR can be sorted properly, but this should work just fine with `alpha<N>` and `beta<N>` phases as long as `N` is a single digit, so I'll let you decide if you want to do that.

If you choose not to bump the release number, we might as well just use `0.` prefix instead of `0.1` if the phase exists.

Reply:
Hmm, good point - I'll take a look.

Reply:
Just FYI, ideally the phase should be part of the version number, and the release should be just `0.N`:
* https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/#_complex_versioning
* https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/#_prerelease_versions

No need to implement this now, I think it will require more testing, but maybe in the future we can properly follow this guideline.

Reply:
I updated the script so now `release number` is:

- No `phase` = 1 (first release)
- `phase` updated = 0.<$CURRENT_RELEASE_NUMBER++>
-  `phase` added = 0.1 (first pre-release)

We can make further modifications later if we decide to change the versioning.

Reply:
> I updated the script so now `release number` is:
> 
> * No `phase` = 1 (first release)
> * `phase` updated = 0.<$CURRENT_RELEASE_NUMBER++>
> * `phase` added = 0.1 (first pre-release)
> 
> We can make further modifications later if we decide to change the versioning.

Hi @edewata, I saw you were OK with this merging but before I do would like to check that this update to the `release_number` makes sense to you, if you would be so kind as to re-review :smile:. Then I will merge and backport etc. Thanks!

Reply:
Yup, LGTM.